network/p2p/blob/blob_service.go

```diff
@@ -232,9 +237,9 @@ func (r *rateLimitedBlockStore) Get(ctx context.Context, c cid.Cid) (blocks.Bloc
 		return nil, err
 	}
 
-	err = r.limiter.WaitN(ctx, size)
-	if err != nil {
-		return nil, err
+	allowed := r.limiter.AllowN(time.Now(), size)
+	if !allowed {
+		return nil, rateLimitedError
```

Comment:
just `return errors.New("rate limited")` here instead of defining a global constant?  you can use `assert.ErrorContains` instead of `assert.ErrorIs`in the test to check for the message